File: client-app/src/app/models/telefoni.ts
import { Profile } from "./profile";

export interface Telefoni {
    id: string;
    emri: string;
    kategoria: string;
    brendi: string;
    data: Date | null;
    pershkrimi: string;
    cmimi: number | null;
    hostUsername: string;
    isCancelled: boolean;
    isGoing: boolean;
    isHost: boolean;
    host?: Profile;
    telefonatPrezencat: Profile[]
}

export class Telefoni implements Telefoni{
    constructor(init?: TelefoniFormValues){
        Object.assign(this, init);
    }
}

export class TelefoniFormValues{
    id?: string = undefined;
    emri: string = '';
    kategoria: string = '';
    brendi: string = '';
    data: Date | null = null;
    pershkrimi: string = '';
    cmimi: number | null = null;

    constructor(telefoni?: TelefoniFormValues){
        if(telefoni){
            this.id = telefoni.id;
            this.emri = telefoni.emri;
            this.kategoria = telefoni.kategoria;
            this.brendi = telefoni.brendi;
            this.data = telefoni.data;
            this.pershkrimi = telefoni.pershkrimi;
            this.cmimi = telefoni.cmimi;
        }
    }
}

File: Domain/Telefoni.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Telefoni
    {
        public Guid Id { get; set; }
        public string Emri { get; set; }
        public string Kategoria { get; set; }
        public string Brendi { get; set; }
        public DateTime Data { get; set; } 
        public string Pershkrimi { get; set; }               
        public decimal Cmimi { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<TelefonatPrezenca> TelefonatPrezencat { get; set; } = new List<TelefonatPrezenca>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}